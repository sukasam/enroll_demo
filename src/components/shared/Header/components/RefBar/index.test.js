import { render, screen, waitFor } from "@testing-library/react";
import { getCookie } from "cookies-next";
import ReferralBar from "./index";

const mockUseUser = jest.fn();
jest.mock("Contexts/UserContext", () => ({
    useUser: () => mockUseUser()
}));

jest.mock("Components/shared/Spinner", () => () => (
    <div data-testid="spinner" />
));
jest.mock("Components/shared/Translate", () => ({ children }) => (
    <span>{children}</span>
));
jest.mock("./styles", () => ({}));

// Mock getCookie
jest.mock("cookies-next", () => ({
    getCookie: jest.fn()
}));

describe("ReferralBar", () => {
    beforeEach(() => {
        jest.clearAllMocks();
        getCookie.mockReturnValue("en");
    });

    it("renders spinner when no enroller name is available", () => {
        mockUseUser.mockReturnValue({
            enrollerFullName: null,
            enrollerAllFullName: null
        });
        render(<ReferralBar />);
        expect(screen.getByTestId("spinner")).toBeInTheDocument();
    });

    it("renders referral information using enrollerFullName when enrollerAllFullName is not available", async () => {
        const enrollerName = "John Doe";
        mockUseUser.mockReturnValue({
            enrollerFullName: enrollerName,
            enrollerAllFullName: null
        });

        render(<ReferralBar />);

        await waitFor(
            () => {
                const enrollerNameElement = screen.getByTestId(
                    "referral_bar_enroller_name"
                );
                expect(enrollerNameElement).toBeInTheDocument();
                expect(enrollerNameElement.textContent.trim()).toBe(
                    enrollerName
                );
            },
            { timeout: 2000 }
        );
    });

    it("renders referral information using language-specific name from enrollerAllFullName", async () => {
        const enrollerName = "John Doe";
        const languageSpecificName = "John Doe (EN)";
        mockUseUser.mockReturnValue({
            enrollerFullName: enrollerName,
            enrollerAllFullName: { "fullName@en": languageSpecificName }
        });

        render(<ReferralBar />);

        await waitFor(
            () => {
                expect(
                    screen.getByTestId("referral_bar_text")
                ).toHaveTextContent("create_account_referred_by");
                expect(
                    screen.getByTestId("referral_bar_enroller_name")
                ).toHaveTextContent(languageSpecificName);
            },
            { timeout: 2000 }
        );
    });

    it("falls back to enrollerFullName when language-specific name is not available", async () => {
        const enrollerName = "John Doe";
        mockUseUser.mockReturnValue({
            enrollerFullName: enrollerName,
            enrollerAllFullName: { "fullName@fr": "John Doe (FR)" }
        });

        render(<ReferralBar />);

        expect(screen.getByTestId("referral_bar_text")).toHaveTextContent(
            "create_account_referred_by"
        );
        expect(
            screen.getByTestId("referral_bar_enroller_name")
        ).toHaveTextContent(enrollerName);
    });

    it("matches snapshot when enroller name is available", async () => {
        const enrollerName = "Jane Smith";
        mockUseUser.mockReturnValue({
            enrollerFullName: enrollerName,
            enrollerAllFullName: { "fullName@en": enrollerName }
        });

        const { container } = render(<ReferralBar />);
        expect(container).toMatchSnapshot();
    });

    it("matches snapshot when no enroller name is available", () => {
        mockUseUser.mockReturnValue({
            enrollerFullName: null,
            enrollerAllFullName: null
        });
        const { container } = render(<ReferralBar />);
        expect(container).toMatchSnapshot();
    });
});
