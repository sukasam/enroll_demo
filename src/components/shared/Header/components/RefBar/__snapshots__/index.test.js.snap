// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferralBar matches snapshot when enroller name is available 1`] = `
<div>
  <div>
    <div
      class="css-0"
    >
      <div
        class="ref-bar-main"
      >
        <span
          class="ref-success"
        >
          <span
            data-testid="referral_bar_text"
          >
            <span>
              create_account_referred_by
            </span>
          </span>
           
          <strong
            data-testid="referral_bar_enroller_name"
          >
            Jane Smith
          </strong>
           
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ReferralBar matches snapshot when no enroller name is available 1`] = `
<div>
  <div>
    <div
      class="css-0"
    >
      <div
        class="ref-bar-main"
      >
        <span
          class="ref-success"
        >
          <div
            data-testid="spinner"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
